tc_driver/dycore.jl

```diff
@@ -201,7 +201,11 @@ function set_grid_mean_from_thermo_state!(param_set, state, grid)
     prog_gm_f = TC.face_prog_grid_mean(state)
     aux_gm = TC.center_aux_grid_mean(state)
     prog_gm_uₕ = TC.grid_mean_uₕ(state)
+
+    @. prog_gm.ρ = TD.air_density(thermo_params, ts_gm)
```

Comment:
```suggestion
```
This doesn't seem correct. We solve for `prog_gm.ρ`.

Reply:
I thought thats what you suggested - to overwrite the dycore state from TC.jl 
I want to do it in the initialization only. The initialisation done in the dycore right now is not compatible with the prescribed profile of Bomex and other single column test cases

Reply:
Ah, yes. Maybe we can put this directly inside the init function though? So that it's not accidentally called inside the RHS?

Reply:
It's part of the list of functions that are being called from `init_tc`: 
https://github.com/CliMA/ClimaAtmos.jl/blob/7a922bcd3c69d959f642c233113e660edf79c9e8/examples/hybrid/TurbulenceConvectionUtils.jl#L116-L150
Plus, we are already overwriting the `prog_gm.rhoe_tot` and prog_gm.rhoq_tot` from that function. 

Are you suggesting the name of the function should be different. Or that it should be explicitly coded inside `init_tc`, without it being a separate function?

Reply:
`set_grid_mean_from_thermo_state` is used only in initialization anyways.